napari/utils/validators.py

```diff
@@ -82,3 +84,34 @@ def func(obj):
                     )
 
     return func
+
+
+def pairwise(iterable: Iterable):
+    """Convert iterable to a zip object containing tuples of pairs along the
+    sequence.
+
+    Example:
+        pairwise(s) == [(s[0], s[1]), (s[1], s[2]), (s[2], s[3]), ...]
+    """
+    # duplicate the iterable
+    a, b = tee(iterable)
+    # shift b by one position
+    next(b, None)
+    # create tuple pairs from the values in a and b
+    return zip(a, b)
+
+
+def validate_increasing(values: Iterable) -> None:
+    """Ensure that values in an iterable are monotocially increasing.
+    Raise exception if `values` is constant or decreasing from one value to
```

Comment:
As a public function, this should include at least one example and should follow the [numpydoc structure](https://numpydoc.readthedocs.io/en/latest/format.html) (e.g. [for `Raises`](https://numpydoc.readthedocs.io/en/latest/format.html#raises)).

Alternatively (and preferred in my opinion), we could declare this to be private (by adding an underscore prefix). And maybe better still just move it in the `intensity_mixin.py`, which is the only place it is used right now.